File: an2_sem4/MPP/Homework/sapt78/java/repo/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/repo/IRepo.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Entity;

public interface IRepo <ID, E extends Entity<ID>> {
    void add(E e) throws EntityRepoException;
    void update(E e) throws EntityRepoException;
    void remove(ID id) throws EntityRepoException;
    E getById(ID id) throws EntityRepoException;
    Iterable<E> getAll() throws EntityRepoException;
}

File: an2_sem4/MPP/Homework/lab3/cs/FestivalSellpoint/FestivalSellpoint.Repo/IRepo.cs
using FestivalSellpoint.Domain;
using System.Collections.Generic;

namespace FestivalSellpoint.Repo
{
    public interface IRepo<ID, E> where E : Entity<ID>
    {
        void Add(E e);
        void Update(E e);
        void Remove(ID id);
        E GetById(ID id);
        IEnumerable<E> GetAll();
    }
}
